### ISSUE TITLE
$mdAria.expectAttribute('aria-label') does not work with interpolated values

### ISSUE THREAD

**Issue by marcysutton** [angular/material#507]
When creating `md-checkbox` and other components with `ng-repeat`, `$mdAria.expectAttribute()` warns about a missing label and does not successfully copy text to the `aria-label` attribute. It seems like a timing issue, because the warnings logged to the console point to nodes in the middle of the transclusion process. See the markup printed to the console, where `md-label` is empty:

![md-checkbox source code in console](https://cloud.githubusercontent.com/assets/1045233/4852899/7d6f8a4e-6080-11e4-8dd2-5ffabf1ca69a.png)

When you click through to the live node, the compiled markup looks like this:

![md-checkbox compiled source](https://cloud.githubusercontent.com/assets/1045233/4852932/140eb1a0-6081-11e4-978e-cbdf8e2950e8.png)

Reprocated in a Plunkr: http://plnkr.co/edit/eD6lgJ3b5HcKKKY0R3VW

**Comment by marcysutton** [comment]
I found this issue occurs on `md-button` when a child element uses `ng-bind`: 

```html
<md-button ng-if="section.type === 'link'"
  ng-class="{'active' : menu.isSectionSelected(section)}"
  ng-href="#{{section.url}}">
  <span ng-bind="section | humanizeDoc"></span>
</md-button>
```
By changing it to interpolate text content directly instead of using `ng-bind`, I was able to suppress the warnings from the console:
```html
<md-button ng-if="section.type === 'link'"
  ng-class="{'active' : menu.isSectionSelected(section)}"
  ng-href="#{{section.url}}">
  {{ section | humanizeDoc }}
</md-button>
```

**Comment by epelc** [comment]
+1 Had this issue with an `md-checkbox` wrapper directive I made.

**Comment by marcysutton** [comment]
@epelc can you post a Codepen or Plunker showing the checkbox issue? With #1420 and work already done on checkboxes I only have the issue on buttons now, since they are fully replaced. So I'd love to see if it's still manifesting in other ways.

**Comment by epelc** [comment]
@marcysutton Do you happen to have a link to one with angular-material setup?

**Comment by epelc** [comment]
@marcysutton http://plnkr.co/edit/DNzMfOve7yEfEXklkAd1?p=preview
There seems to be problems with using `<md-checkbox>{{SomeValue}}</md-checkbox>` without also adding `aria-label="{{SomeValue}}"`

**Comment by ThomasBurleson** [comment]
@epelc  - In your Plunkr, I am not seeing **any** console out for aria warnings. Any thoughts?

**Comment by epelc** [comment]
@ThomasBurleson I think it got fixed in master
